otx/api/usecases/exportable_code/visualizers/visualizer.py

```diff
@@ -111,3 +119,48 @@ def is_quit(self) -> bool:
             return False
 
         return ord("q") == cv2.waitKey(self.delay)
+
+    def save_frame(self, image, input_path, streamer_type) -> None:
+        """Save result image into dict.
+
+        Args:
+            image (np.ndarray): Image to be saved.
+            input_path (str): Filename of the image
+            streamer_type (str) : The type of the input
+        """
+
+        if self.save_results_to and input_path:
+            filename = Path(input_path).name
+            if filename not in self.saved_frames:
+                self.saved_frames[filename] = []
+            if "VIDEO" in streamer_type:
+                self.saved_frames[filename].append(image)
+            else:
+                self.saved_frames[filename] = image
+
+    def dump_frames(self, streamer) -> None:
+        """Save frames to file system.
+
+        Args:
+            streamer (str): The streamer with images to be saved
+        """
+        if len(self.saved_frames) > 0 and self.save_results_to:
```

Comment:
```suggestion
        if self.saved_frames and self.save_results_to:
```
Although It doesn't matter, this way is recommend to check whether dict is empty :)

Reply:
Thank you :)